src/capplet/mate-notification-applet.c

```diff
@@ -0,0 +1,216 @@
+/* mate-notification-applet.c - MATE Notification Applet
+ *
+ * Copyright (c) 2020 Robert Buj <robert.buj@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+#include "config.h"
+
+#include <glib.h>
+#include <glib/gi18n.h>
+#include <gtk/gtk.h>
+#include <mate-panel-applet.h>
+
+#define MATE_DESKTOP_USE_UNSTABLE_API
+#include <libmate-desktop/mate-desktop-utils.h>
+
+#define GSETTINGS_SCHEMA "org.mate.NotificationDaemon"
+#define GSETTINGS_KEY_DO_NOT_DISTURB "do-not-disturb"
+
+typedef struct
+{
+  MatePanelApplet *applet;
+
+  GtkWidget       *image_on;
+  GtkWidget       *image_off;
+  GtkActionGroup  *action_group;
+  GSettings       *settings;
+} MateNotificationApplet;
+
+static void
+show_about      (GtkAction              *action,
+                 MateNotificationApplet *applet);
+static void
+call_properties (GtkAction              *action,
+                 MateNotificationApplet *applet);
+
+static const GtkActionEntry applet_menu_actions [] = {
+  { "Preferences", "document-properties", N_("_Preferences"),
+    NULL, NULL, G_CALLBACK (call_properties) },
+  { "About", "help-about", N_("_About"),
+    NULL, NULL, G_CALLBACK (show_about) }
+};
+
+static const char* notification_properties[] = {
+  "mate-notification-properties",
+};
+
+static void
+applet_destroy (MatePanelApplet        *applet_widget,
+                MateNotificationApplet *applet)
+{
+  g_assert (applet);
+
+  g_object_unref (applet->settings);
+  g_object_unref (applet->action_group);
+  g_free (applet);
+}
+
+static void
+call_properties (GtkAction              *action,
+                 MateNotificationApplet *applet)
+{
+  gsize i;
+
+  for (i = 0; i < G_N_ELEMENTS (notification_properties); i++) {
+    char *programpath = g_find_program_in_path (notification_properties[i]);
+
+    if (programpath != NULL) {
+      g_free (programpath);
+      mate_gdk_spawn_command_line_on_screen (gtk_widget_get_screen (GTK_WIDGET (applet->applet)),
+                                             notification_properties[i],
+                                             NULL);
+      return;
+    }
+  }
+}
+
+static void
+show_about (GtkAction              *action,
+            MateNotificationApplet *applet)
```

Comment:
https://travis-ci.org/github/mate-desktop/mate-notification-daemon/jobs/744844272#L2254
```
mate-notification-applet.c:93:37: warning: unused parameter 'applet' [-Wunused-parameter]

            MateNotificationApplet *applet)
```

I am still wondering why both parameters are unused 0_o

Reply:
It's a callback method.